packages/serverless/src/integrations/aws_lambda.ts

```diff
@@ -0,0 +1,235 @@
+import { EventProcessor, Hub, Integration, Scope } from '@sentry/types';
+import * as Sentry from '@sentry/node';
+
+/**
+ * NodeJS integration
+ *
+ * Provides a mechanism for NodeJS
+ * that raises an exception for handled, unhandled, timeout and similar times of error
+ * and captures the same in Sentry Dashboard
+ */
+export class AWSLambda implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'AWSLambda';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = AWSLambda.id;
+
+  /**
+   * Returns current hub
+   */
+  private getCurrentHub?: () => Hub;
+
+  /**
+   * context.
+   */
+  private awsContext: any;
+
+  /**
+   * timeout flag.
+   */
+  private timeoutWarning: boolean = false;
+
+  /**
+   * configured time in miliseconds.
+   */
+  private configuredTimeInMilliseconds: number;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(
+    options: { awslambda?: any; context?: any; timeoutWarning?: any; configuredTimeInMilliseconds?: number } = {},
+  ) {
+    this.awsContext = options.context;
+    this.timeoutWarning = options.timeoutWarning;
+    this.configuredTimeInMilliseconds = options.configuredTimeInMilliseconds as number;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    // lambda-bootstrap
+    const lambdaBootstrap = require.main;
+
+    if (!this.awsContext && !lambdaBootstrap) {
+      return;
+    }
+
+    const nodeJs: string = 'node';
+
+    /** capturing this context */
+    const setupOnceContext = this;
```

Comment:
You are inside the class. There's no need to store the reference to self anywhere.

Reply:
made this change in the new pull request.